source/github/webhook.go

```diff
@@ -259,6 +261,22 @@ func processDeploymentEvent(h *library.Hook, payload *github.DeploymentEvent) (*
 	b.SetBranch(payload.GetDeployment().GetRef())
 	b.SetRef(payload.GetDeployment().GetRef())
 
+	// check if payload is provided within request
+	if payload.GetDeployment().Payload != nil {
+		deployPayload := raw.StringSliceMap{}
+		// unmarshal the payload into the expected map[string]string format
+		err := json.Unmarshal(payload.GetDeployment().Payload, &deployPayload)
+		if err != nil {
+			return nil, err
```

Comment:
Thoughts on if we should skip the payload data if it doesn't match `map[string]string` instead of returning some error?

Reply:
Can a deployment contain an empty payload?
Is it possible the deploy payload value could just be a string or a nil value that isnt {}? In the event of an error, etc?
If so, I think skipping when it doesnt match on map[string]string is fine, but if deployment is expected to _always_ have an error I like checking for unmarshal errors.

Reply:
I inquired with GitHub support about the payload and they responded with: “If the payload is passed in the input when creating a deployment, this endpoint expects it to either be a JSON object or a string. Otherwise, a validation error will be thrown.“. I'll play around with creating some deployments directly with the GitHub API and add relevant tests to handle those scenarios.